airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py

```diff
@@ -27,19 +28,23 @@ def __init__(
         name: str,
         url_base: [str, InterpolatedString],
         path: [str, InterpolatedString],
-        http_method: Union[str, HttpMethod],
-        request_options_provider: RequestOptionsProvider = None,
-        request_headers_provider: RequestHeaderProvider = None,
+        http_method: Union[str, HttpMethod] = HttpMethod.GET,
+        request_options_provider: Optional[RequestOptionsProvider] = None,
+        request_headers_provider: Optional[RequestHeaderProvider] = None,
```

Comment:
@brianjlai do you think it would make sense to move the request headers in the `RequestOptionsProvider`?